loadtest/control/gencontroller/controller.go

```diff
@@ -137,6 +126,21 @@ func (c *GenController) Run() {
 			frequency:  int(math.Ceil(float64(c.config.NumChannels) * c.config.PercentGroupChannels)),
 			idleTimeMs: 1000,
 		},
+	}
+
+	c.runActions(actions, func() bool { return st.get("channels") >= c.config.NumChannels })
+
+	actions = map[string]userAction{
+		"joinTeam": {
+			run:        control.JoinTeam,
+			frequency:  100,
+			idleTimeMs: 0,
+		},
+		"joinChannel": {
+			run:        c.joinChannel,
+			frequency:  int(math.Ceil(float64(c.config.NumChannels))) * 2, // making this proportional to number of channels.
```

Comment:
I suspect this could become crazy frequent this way. May need some further review after testing a bit.

Also following a similar pattern to the rest of the actions we may want to prevent running this once everyone has joined.

Reply:
NumChannels is usually orders of magnitude lesser than NumPosts. In my testing, this was executed way too rarely with high post count. That's why I made this to be proportional to channels.

> Also following a similar pattern to the rest of the actions we may want to prevent running this once everyone has joined.

It is bounded by the memberLimit of a given channel.

Reply:
> It is bounded by the memberLimit of a given channel.

But how do we enforce this? I am struggling to understand how we decide when enough users have joined.